aspnetcore/host-and-deploy/health-checks/samples/2.x/HealthChecksSample/Services/StartupHostedService.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace SampleApp.Services
+{
+    #region snippet1
+    public class StartupHostedService : IHostedService, IDisposable
+    {
+        private readonly int _delaySeconds = 15;
+        private readonly ILogger _logger;
+        private readonly StartupHostedServiceHealthCheck _startupHostedServiceHealthCheck;
+
+        public StartupHostedService(ILogger<StartupHostedService> logger, 
+            StartupHostedServiceHealthCheck startupHostedServiceHealthCheck)
+        {
+            _logger = logger;
+            _startupHostedServiceHealthCheck = startupHostedServiceHealthCheck;
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"Startup Background Service is starting.");
+
+            Task.Delay(_delaySeconds * 1000)
+                .ContinueWith(_ => 
+                    { _startupHostedServiceHealthCheck.StartupTaskCompleted = true; });
```

Comment:
You could also do something like:
```
Task.Run(async () =>
{
    // This is intended to simulate the effect of a long-running startup task.
    await Task.Delay(_delaySeconds * 1000);

    _startupHostedServiceHealthCheck.StartupTaskCompleted = true;
});
```

This is a little more idiomatic.

Reply:
humm ... I thought that doing it that way would block the whole app from starting. Maybe not ... I trust *you* a LOT more than I trust *silly green dinosaurs*. lol

Reply:
yeah :) it returns a task, but doesn't block itself. https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.run?view=netframework-4.7.2#System_Threading_Tasks_Task_Run_System_Func_System_Threading_Tasks_Task__